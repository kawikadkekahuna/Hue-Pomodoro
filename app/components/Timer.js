import glamorous from "glamorous";
import React, { Component } from "react";

const { H1, Div } = glamorous;

export default class Timer extends Component {
  constructor(props) {
    super(props);
  }

  getMinutes(seconds) {
    return Math.floor(seconds / 60);
  }

  getSeconds(totalSeconds) {
    return totalSeconds % 60 ? totalSeconds % 60 : "00";
  }

  render() {
    return (
      <Div
        className="timer-group"
        display="flex"
        justifyContent="center"
        alignItems="center"
      >
        <Div className="face">
          <H1 fontSize="20em" id="lazy">
            {this.getMinutes(this.props.time)}
            :
            {this.getSeconds(this.props.time)}
          </H1>
        </Div>
      </Div>
    );
  }
}
