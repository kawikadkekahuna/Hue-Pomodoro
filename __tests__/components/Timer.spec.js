import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Timer from '../../app/components/Timer';

describe('Timer Component', () => {
  test('has correct default state', () => {
    const tree = shallow(<Timer />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  test('correctly returns seconds text', () => {
    const tree = shallow(<Timer />);
    expect(tree.instance().getSeconds(60)).toEqual('00');
    expect(tree.instance().getSeconds(59)).toEqual(59);
  });
});
