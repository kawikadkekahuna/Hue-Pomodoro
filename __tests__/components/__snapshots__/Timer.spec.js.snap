// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pomodoro Component can handle pomodoro event 1`] = `
<div>
  <glamorous.Div
    position="absolute"
    right="0"
    top="0"
  >
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/hue-config"
    >
      <ActionSettings />
    </Link>
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/hue-config"
    >
      <ActionLightbulbOutline />
    </Link>
  </glamorous.Div>
  <glamorous.Div>
    <Timer />
    <glamorous.Div
      alignItems="center"
      display="flex"
      justifyContent="center"
    >
      <Button
        className="pomodoroToggle"
        onClick={[Function]}
      />
      <Button
        text="Reset"
      />
    </glamorous.Div>
  </glamorous.Div>
</div>
`;

exports[`Pomodoro Component has correct default state 1`] = `
<div>
  <glamorous.Div
    position="absolute"
    right="0"
    top="0"
  >
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/hue-config"
    >
      <ActionSettings />
    </Link>
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/hue-config"
    >
      <ActionLightbulbOutline />
    </Link>
  </glamorous.Div>
  <glamorous.Div>
    <Timer />
    <glamorous.Div
      alignItems="center"
      display="flex"
      justifyContent="center"
    >
      <Button
        className="pomodoroToggle"
        onClick={[Function]}
      />
      <Button
        text="Reset"
      />
    </glamorous.Div>
  </glamorous.Div>
</div>
`;

exports[`Timer Component correctly returns seconds text 1`] = `
<glamorous.Div
  alignItems="center"
  className="timer-group"
  display="flex"
  justifyContent="center"
>
  <glamorous.Div
    className="face"
  >
    <glamorous.H1
      fontSize="20em"
      id="lazy"
    >
      NaN
      :
      00
    </glamorous.H1>
  </glamorous.Div>
</glamorous.Div>
`;

exports[`Timer Component has correct default state 1`] = `
<glamorous.Div
  alignItems="center"
  className="timer-group"
  display="flex"
  justifyContent="center"
>
  <glamorous.Div
    className="face"
  >
    <glamorous.H1
      fontSize="20em"
      id="lazy"
    >
      NaN
      :
      00
    </glamorous.H1>
  </glamorous.Div>
</glamorous.Div>
`;
